build/phase.go

```diff
@@ -148,3 +147,21 @@ func (p *Phase) Run(context context.Context) error {
 func (p *Phase) Cleanup() error {
 	return p.docker.ContainerRemove(context.Background(), p.ctr.ID, types.ContainerRemoveOptions{Force: true})
 }
+
+func (p *Phase) createAppReader() (io.ReadCloser, error) {
```

Comment:
Similar to my comments on the original buildpack tgz PR, it would be nice if the different formats of input parameters were abstracted away from lower level components. Given the `client.Build` handles validating inputs, perhaps it can provide the Lifecycle with an App struct of some sort that the Phase can just read from (so we don't need to think about file types in many different places). I am okay merging this w/o making that change now but I would like to drive toward that design in the future.

Reply:
I like that idea. Agree that it's not 100% necessary, but that we should do it soon. I'll make a chore for it.

Reply:
Chore link: https://github.com/orgs/buildpack/projects/1#card-23809969